app.js

```diff
@@ -103,34 +103,24 @@ function updateMetronome() {
 let isPlaying = false;
 let timeout;
 
+let lastClick = performance.now();
 function playMetronome() {
-  console.log("click");
+  const now = performance.now();
+  const expected = 60000 / tempo;
+  const actual = now - lastClick;
+  const diff = new Intl.NumberFormat("en-US", { signDisplay: "exceptZero" }).format(actual - expected);
+  console.log("click", { expected, actual, diff });
+  lastClick = now;
```

Comment:
Try running this and seeing what happens on your machine. I see sometimes a lag of 10 - 30 ms, sometimes only a couple of millis. That it's consistent is good. That it slows down the actual tempo is bad.

I'd like you to try and resolve this. Let me know if you need some hints!

Don't put off getting clicks playing to work on this. They're orthogonal.

Reply:
I will work on this today and tomorrow. Thanks!